bft-lib/src/base_types.rs

```diff
@@ -1,20 +1,19 @@
 // Copyright (c) Calibra Research
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::Error;
 use serde::{Deserialize, Serialize};
 use std::fmt;
 
 #[cfg(test)]
 #[path = "unit_tests/base_type_tests.rs"]
 mod base_type_tests;
 
-// TODO: add error handling (using thiserror to define an error type?) + remove Unpin
-// Alternatively, we may want to use a generic associated type when there are available on
-// rust-stable:   https://github.com/rust-lang/rust/issues/44265
-pub type AsyncResult<T> = Box<dyn std::future::Future<Output = T> + Unpin + 'static>;
+// TODO: better error type
+pub type Result<T> = std::result::Result<T, anyhow::Error>;
 
-pub type Result<T> = std::result::Result<T, Error>;
+pub type Async<'a, T> = futures::future::BoxFuture<'a, T>;
+
+pub type AsyncResult<'a, T> = futures::future::BoxFuture<'a, Result<T>>;
```

Comment:
Is this an attempt to restrict the very permissive type of async-await?

Reply:
The previous version was indeed strict and not very usable.